src/applications/personalization/profile/components/personal-information/ContactInformationActionButtons.jsx

```diff
@@ -7,138 +7,150 @@ import { toLower } from 'lodash';
 
 import recordEvent from '~/platform/monitoring/record-event';
 import LoadingButton from '~/platform/site-wide/loading-button/LoadingButton';
+import { usePrevious } from '~/platform/utilities/react-hooks';
+
+function AlertContent({ cancelAction, deleteAction, isLoading, title }) {
+  return (
+    <div>
+      <p>
+        This will delete your {toLower(title)} across many VA records. You can
+        always come back to your profile later if you’d like to add this
+        information back in.
+      </p>
+      <div>
+        <LoadingButton isLoading={isLoading} onClick={deleteAction}>
+          Confirm
+        </LoadingButton>
+
+        {!isLoading && (
+          <button
+            type="button"
+            className="usa-button-secondary"
+            onClick={cancelAction}
+          >
+            Cancel
+          </button>
+        )}
+      </div>
+    </div>
+  );
+}
 
-class ContactInformationActionButtons extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      deleteInitiated: false,
-    };
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    // Once the AlertBox is mounted, we want to set the focus to the heading
-    // for screen reader use
-    if (this.state.deleteInitiated && !prevState.deleteInitiated) {
-      const heading = document.getElementById('deleteConfirmationHeading');
-      if (heading) {
-        heading.focus();
+function ContactInformationActionButtons(props) {
+  const [deleteInitiated, setDeleteInitiated] = React.useState(false);
+  const [focusDeleteAlert, setFocusDeleteAlert] = React.useState(false);
+  const wasDeleteInitiated = usePrevious(deleteInitiated);
+
+  React.useEffect(
+    () => {
+      if (deleteInitiated && !wasDeleteInitiated) {
+        const heading = document.getElementById('deleteConfirmationHeading');
+        if (heading) {
+          heading.focus();
+        }
       }
-    }
-    // If delete is cancelled, put focus back on modal close button
-    if (!this.state.deleteInitiated && prevState.deleteInitiated) {
-      const closeButton = document.getElementsByClassName('va-modal-close')[0];
-      if (closeButton) {
-        closeButton.focus();
+      if (!deleteInitiated && wasDeleteInitiated) {
+        const closeButton = document.getElementsByClassName(
+          'va-modal-close',
+        )[0];
+        if (closeButton) {
+          closeButton.focus();
+        }
       }
-    }
-  }
+    },
+    [deleteInitiated, wasDeleteInitiated],
+  );
+
+  React.useEffect(
```

Comment:
Had to read your through your changes a few times to understand what part fixed the issue, but I think I understand now! 💯

Reply:
Yeah it's a bit hacky, perhaps. This might call for a code comment explaining what's going on? But `focusDeleteAlert` is set to true as soon as we start the deletion process. And `focusDeleteAlert` being `true` results in the `AlertBox` scrolling into view when it's added to the DOM. But then we immediately set `focusDeleteAlert` to `false` so that the `AlertBox` doesn't keep scrolling into view if the user happens to change data in the form.